internal/pkg/etcd/etcd.go

```diff
@@ -148,53 +149,75 @@ func validateMemberHealth(ctx context.Context, memberURIs []string) (err error)
 }
 
 // LeaveCluster removes the current member from the etcd cluster and nukes etcd data directory.
-func (c *Client) LeaveCluster(ctx context.Context) error {
-	hostname, err := os.Hostname()
-	if err != nil {
-		return err
-	}
-
-	if err = c.RemoveMember(ctx, hostname); err != nil {
+func (c *Client) LeaveCluster(ctx context.Context, memberID string) error {
+	if err := c.RemoveMemberByMemberIDStr(ctx, memberID); err != nil {
 		return err
 	}
 
-	if err = system.Services(nil).Stop(ctx, "etcd"); err != nil {
+	if err := system.Services(nil).Stop(ctx, "etcd"); err != nil {
 		return fmt.Errorf("failed to stop etcd: %w", err)
 	}
 
 	// Once the member is removed, the data is no longer valid.
-	if err = os.RemoveAll(constants.EtcdDataPath); err != nil {
+	if err := os.RemoveAll(constants.EtcdDataPath); err != nil {
 		return fmt.Errorf("failed to remove %s: %w", constants.EtcdDataPath, err)
 	}
 
 	return nil
 }
 
-// RemoveMember removes the member from the etcd cluster.
-func (c *Client) RemoveMember(ctx context.Context, hostname string) error {
+func (c *Client) getMemberID(ctx context.Context, hostname string) (uint64, error) {
 	resp, err := c.MemberList(ctx)
 	if err != nil {
-		return err
+		return 0, err
 	}
 
-	var id *uint64
-
 	for _, member := range resp.Members {
 		if member.Name == hostname {
 			member := member
-			id = &member.ID
 
-			break
+			return member.ID, nil
 		}
 	}
 
-	if id == nil {
-		return fmt.Errorf("failed to find %q in list of etcd members", hostname)
+	return 0, fmt.Errorf("could not get member ID for hostname %q", hostname)
+}
+
+// RemoveMemberByHostnameOrMemberID removes the member from the etcd cluster.
+func (c *Client) RemoveMemberByHostnameOrMemberID(ctx context.Context, hostnameOrMemberID string) error {
+	var id uint64
+
+	id, err := etcdresource.MemberIDToUint64(hostnameOrMemberID)
+	if err != nil {
+		id, err = c.getMemberID(ctx, hostnameOrMemberID)
+		if err != nil {
+			return err
+		}
 	}
 
-	_, err = c.MemberRemove(ctx, *id)
+	err = c.RemoveMemberByMemberID(ctx, id)
 	if err != nil {
-		return fmt.Errorf("failed to remove member %d: %w", *id, err)
+		return fmt.Errorf("failed to remove member %d: %w", id, err)
+	}
+
+	return nil
+}
+
+// RemoveMemberByMemberIDStr removes the member from the etcd cluster.
+func (c *Client) RemoveMemberByMemberIDStr(ctx context.Context, memberID string) error {
```

Comment:
I feel like we should have explicit methods for member ID/hostname, it feels much better.